### ISSUE TITLE
Invalid drawio url for private gitlab and publish url

### ISSUE THREAD

**Issue by rickywu** [jgraph/drawio#515]
Two problem here

1. If draw.io run in sub directory, such as http://mycompany.net/draw and use private gitlab, after auth, the redirect url just use window.location.origin + '/gitlab.html'. see https://github.com/jgraph/drawio/blob/bfd688d0ba48f8a24bb5fe69c90605c2725a7e06/src/main/webapp/js/diagramly/GitLabClient.js#L42

    This cause invalid redirect url problem in gitlab, I think it should also setup env variable sich as window.DRAWIO_URL='http://mycompany.net/draw' + '/gitlab.html'

2. Publish as url still use draw.io, the same problem, probally should setup in config file
![image](https://user-images.githubusercontent.com/442398/62603440-7f795800-b928-11e9-9b35-a83e56d61bce.png)

**Comment by m4r10k** [comment]
Yes, this would be really nice because otherwise the "root" context of a domain is used up by drawio. Furthermore in a lot of configurations some kind of layer 7 load balancer is used usually to switch different applications contexts behind a domain name. Therefore it would be helpful if drawio could be deployed as normal application inside the Tomcat (not root).

**Comment by bsima** [comment]
The GitLab URL param doesn't seem to be respected at all. I have this nginx config:

```
    server {
        listen 10.0.0.1:80; # service-1
        server_name draw.internal.example.com;
        location / {
            # draw.io does it's configuration via URL parameters...
            # https://desk.draw.io/support/solutions/articles/16000042546-what-url-parameters-are-supported-
            set $args gapi=0&db=0&od=0&tr=0&gh=0&browser=0&gitlab=https%3A%2F%2Fgitlab.example.com&gitlab-id=[redacted];
            proxy_pass http://127.0.0.1:3001;
        }
    }
```

And yet it still tries `gitlab.com`...

**Comment by davidjgraph** [comment]
When does the call to gitlab.com go out. I tried:

`https://www.draw.io/?gitlab=https%3A%2F%2Fgitlab.example.com&gitlab-id=[redacted]`

(yes, I know it won't work), but clicking on authorize sent me to:

`https://gitlab.example.com/oauth/authorize?client_id=[redacted]&scope=api%20read_repository%20write_repository&redirect_uri=https%3A%2F%2Fwww.draw.io%2Fgitlab.html&response_type=token&state=123`

That looks right to me.

**Comment by davidjgraph** [comment]
1 fixed in 12.2.1

As for part 2, publish includes the diagram data, a custom Gitlab install doesn't affect that. If you've a custom draw.io install and that's the publish URL, that it is a bug, but please log another ticket for that.
